src/Shell/BackupDatabaseShell.php

```diff
@@ -52,9 +52,10 @@ public function main()
         $configFileObject = new File($configFile);
         $configFileContent = '[mysqldump]
 host=%host%
+port=%port%
 user=%user%
 password="%password%"';
-        $configFileContent = str_replace(['%host%', '%user%', '%password%'], [$dbConfig['host'], $dbConfig['username'], $dbConfig['password']], $configFileContent);
+        $configFileContent = str_replace(['%host%', '%port%', '%user%', '%password%'], [$dbConfig['host'], $dbConfig['port'], $dbConfig['username'], $dbConfig['password']], $configFileContent);
```

Comment:
Without providing a port you get a `Warning Error: Undefined array key "port"` in line 58.
If you fix that with an "isset check" I'll merge your PR.
Thank you,
Mario